### PR TITLE
DOC: Improved documentation for diag_indices and diag_indices_from 

### PR DESCRIPTION
**Pull request by selamw1**
Part of https://github.com/google/jax/issues/21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -5248,8 +5270,35 @@ def diag_indices(n: int, ndim: int = 2) -> tuple[Array, ...]:
...
+  Args:
+    arr: Input array. Must be at least 2-dimensional and have equal length along
+      all dimensions.
+
+  Returns:
+    A tuple of arrays containing the indices to access the main diagonal of
+    the input array.
+
+  See also:
+    - :func:`jax.numpy.diag_indices`
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:5287]: Also reference `jax.numpy.diagonal`, which can be used to extract dieagonal values from an array.

**Review by jakevdp — COMMENTED** [review]
Looks great! Just a couple minor comments

**Comment by selamw1** [comment]
Thanks @jakevdp! it's done.

**Review by jakevdp — APPROVED** [review]
Looks good, thanks!

**Pull request merged**